--- CMakeLists.txt	2025-11-28 16:31:04.994521792 +0800
+++ ../ftbl/fightable/CMakeLists.txt	2025-12-01 20:55:26.931894722 +0800
@@ -77,6 +77,19 @@
     ext/args/src/args.c
 )
 
+include(CheckIncludeFile)
+
+find_library(CURL_LIB curl)
+if(NOT CURL_LIB)
+    message(FATAL_ERROR "libcurl could not be found")
+endif()
+CHECK_INCLUDE_FILE("curl/curl.h" HAVE_LIBCURL)
+if (NOT HAVE_LIBCURL AND NOT CURL_INC)
+    message(FATAL_ERROR "libcurl binaries were found but the header files itself are missing")
+endif()
+message(STATUS "curl custom include path: ${CURL_INC}")
+
+
 add_library(fshared STATIC
     src/shared/src/cstr.c
     src/shared/src/uchar.c
@@ -96,9 +109,9 @@
     src/shared/src/int.c
 )
 target_include_directories(fshared PRIVATE
-    ${FIGHTABLE_SHARED_INC}
+    ${FIGHTABLE_SHARED_INC} ${CURL_INC}
 )
-target_link_libraries(fshared curl)
+target_link_libraries(fshared ${CURL_LIB})
 
 if (LIBC_FD_SIZE)
     message(STATUS "Setting MP stack definition, FD_SETSIZE, to ${LIBC_FD_SIZE}")
@@ -116,8 +129,6 @@
     set(MICROHTTPD_LIB_CUSTOM TRUE)
 endif()
 
-include(CheckIncludeFile)
-
 find_library(OPENMPT_LIB openmpt)
 if (NOT OPENMPT_LIB)
     message(FATAL_ERROR "libopenmpt could not be found")
@@ -144,11 +155,6 @@
 endif()
 
 if (NOT DISABLE_MP_SERVER)
-    find_library(CURL_LIB curl)
-    if(NOT CURL_LIB)
-        message(FATAL_ERROR "libcurl could not be found")
-    endif()
-
     find_library(MICROHTTPD_LIB microhttpd)
     if(NOT MICROHTTPD_LIB)
         message(FATAL_ERROR "libmicrohttpd could not be found")
@@ -156,11 +162,6 @@
 
     message(STATUS "libmicrohttpd lib path: ${MICROHTTPD_LIB}")
 
-    CHECK_INCLUDE_FILE("curl/curl.h" HAVE_LIBCURL)
-    if (NOT HAVE_LIBCURL)
-        message(FATAL_ERROR "libcurl binaries were found but the header files itself are missing")
-    endif()
-
     CHECK_INCLUDE_FILE("microhttpd.h" HAVE_LIBMICROHTTPD)
     if (NOT HAVE_LIBMICROHTTPD AND NOT MICROHTTPD_INC)
         message(FATAL_ERROR "libmicrohttpd binaries were found but the header files itself are missing")
@@ -202,7 +203,12 @@
 
 if (NOT DISABLE_PORTAUDIO)
     set(WDMKS OFF)
-    add_subdirectory(ext/portaudio_opensles)
+    if (NOT LINUX)
+        set(PA_USE_ALSA OFF)
+        set(PA_USE_PULSEAUDIO OFF)
+        set(PA_USE_OSS ON)
+    endif()
+    add_subdirectory(ext/portaudio)
 else()
     add_compile_definitions(_DISABLE_PORTAUDIO_)
 endif()
@@ -621,7 +627,7 @@
     src/client/inc
     ${FIGHTABLE_RENDERER_INC}
     ext/cJSON
-    ext/portaudio_opensles/include
+    ext/portaudio/include
     ext/nativefiledialog-extended/src/include
     ext/PerlinNoise
     ext/nbnet
@@ -658,7 +664,7 @@
     ${FIGHTABLE_NATIVE_INCLUDES}
 )
 if (NOT DISABLE_PORTAUDIO)
-    target_link_libraries(fbase ${FIGHTABLE_RENDERER_LIB} cjson PortAudio ${OPENMPT_LIB} ${FIGHTABLE_NATIVE_LIB} ${FIGHTABLE_HTTP_LINK} fshared args)
+    target_link_libraries(fbase ${FIGHTABLE_RENDERER_LIB} cjson portaudio ${OPENMPT_LIB} ${FIGHTABLE_NATIVE_LIB} ${FIGHTABLE_HTTP_LINK} fshared args)
 else()
     target_link_libraries(fbase ${FIGHTABLE_RENDERER_LIB} cjson ${OPENMPT_LIB} ${FIGHTABLE_NATIVE_LIBS} ${FIGHTABLE_HTTP_LINK} fshared args)
 endif()
